posthog/hogql/grammar/HogQLParser.g4

```diff
@@ -37,8 +36,7 @@ groupByClause: GROUP BY ((CUBE | ROLLUP) LPAREN columnExprList RPAREN | columnEx
 havingClause: HAVING columnExpr;
 orderByClause: ORDER BY orderExprList;
 projectionOrderByClause: ORDER BY columnExprList;
-limitByClause: LIMIT limitExpr BY columnExprList;
-limitClause: LIMIT limitExpr (WITH TIES)?;
+limitClause: LIMIT limitExpr ((WITH TIES) | BY columnExprList)?;
```

Comment:
I checked and CH doesn't support BOTH a "with ties" and a "by x" ending on "limit" at the same time, though the parser seemed to allow it. Fixed.